src/main/game_loop.cpp

```diff
@@ -88,19 +107,19 @@ void GameLoop()
         UpdateEffects();
         speedRun_tick();
         UpdateGraphics();
+        updateScreenFaders();
```

Comment:
Can you put these calls in `UpdateGraphics` instead? Maybe more maintainable to have fewer special calls in the main loops

Reply:
This thing is a backend part, and I think it's better to separate render and other workflow-related updates

Reply:
I think the screen faders are similar to the screen shake, which does live in `UpdateControls`, but if you prefer it here I am good with that also.

Reply:
I think we could try something later to tweak this and clean up the whole stuff: in the future, I plan to reorganize the whole rendering making it use the Z-value concept, which should allow the flexible rendering of objects like at Moondust and at X2 is possible. This thing is required for a multi-layered backgrounds system which already requires some Z-value support to allow drawing layers in many ways:
- under everything like usually now
- among in-scene objects
- over everything (making sort of decorations like plants, etc.)

Therefore, this thing could be improved later :wink:

Reply:
This sounds exciting, and you can see in the devel branch one approach to separate the logic components of `UpdateGraphics` from the actual rendering. The NPC draw queue class is basically a simple (and high performance, since no sorting is required) Z-value system. If we use a sorting-based system instead of a multi-queue system we should be careful to ensure high performance on the systems like Vita and 3DS that still need optimization.

Reply:
So, that's a topic for future discussing :wink: